src/App/_tests_/ratings.test.js

```diff
@@ -0,0 +1,221 @@
+import React from 'react';
+import configureStore from 'redux-mock-store';
+import thunk from 'redux-thunk';
+import promiseMiddleware from 'redux-promise-middleware';
+import { shallow } from '../../enzyme';
+import { Ratings, mapStateToProps } from '../Components/DisplayArticle/Ratings';
+import reducer from '../../Redux/Reducers/articleRatings';
+import mockAxios from '../../configs/axios';
+import actions from '../../Redux/Actions/articleRatings';
+
+const middlewares = [thunk, promiseMiddleware];
+const mockStore = configureStore(middlewares);
+
+const initialState = {
+  articleRatings: {
+    data: { allUsers: 13, statistics: [{ rating: 1, users: [], percentage: 3.2 }] },
+    errorMessage: 'message',
+    successMessage: 'message'
+  }
+};
+const store = mockStore(initialState);
+
+mapStateToProps(initialState);
+
+const props = { getRatings: () => jest.fn(), rate: () => Promise.resolve({}) };
+
+const component = shallow(<Ratings {...props} />);
+describe('Ratings', () => {
+  it('renders the Ratings component', () => {
+    component.instance().componentWillReceiveProps({
+      ratings: {
+        average: '3.2',
+        data: { allUsers: 13, statistics: [{ rating: 1, users: [], percentage: 3.2 }] }
+      },
+      errorMessage: 'empty',
+      successMessage: 'success'
+    });
+
+    component.instance().render();
+    expect(component.find('Fragment').exists()).toEqual(true);
+  });
+
+  it('Renders the toast when there is no token', () => {
+    component.setState({ redirect: false });
+    component.instance().rate(1);
+    expect(component.find('ToastContainer').exists()).toEqual(true);
+  });
+
+  it('Uses the token to rate', () => {
+    // Fires with the token
+    localStorage.setItem('token', 'token');
+    component.setState({ redirect: false });
+    component.instance().rate(1);
+    expect(component.find('ToastContainer').exists()).toEqual(true);
+    expect(component.state().rating).toEqual(1);
+  });
+
+  it('Alerts when the rating is the same', () => {
+    // Fires with the token
+    localStorage.setItem('token', 'token');
+    component.setState({ redirect: false });
+    component.instance().rate(1);
+    expect(component.find('ToastContainer').exists()).toEqual(true);
+  });
+});
+
+describe('Actions', () => {
+  const ratingsMock = {
+    status: 200,
+    data: {
+      allUsers: 37,
+      statistics: [
+        {
+          rating: 1,
+          users: 7,
+          percentage: 19
+        },
+        {
+          rating: 3,
+          users: 9,
+          percentage: 24
+        },
+        {
+          rating: 5,
+          users: 7,
+          percentage: 19
+        },
+        {
+          rating: 2,
+          users: 4,
+          percentage: 11
+        },
+        {
+          rating: 4,
+          users: 10,
+          percentage: 27
+        }
+      ]
+    }
+  };
+
+  const rateArticleError = {
+    status: 401,
+    data: { errors: { ratings: 'you cannot rate twice this article, thanks' } }
```

Comment:
@nkpremices is it    ***a user can not rate an `article twice`*** or ***a user can not rate an `article twice with the same rate`***?